apps/beeai-web/src/acp/api.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2025 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { unstable_cache } from "next/cache";
+import { getAcpClient } from "./client";
+
+export const getAgentsList = unstable_cache(
+  async () => {
+    const client = await getAcpClient();
+
+    const { agents } = await client.listAgents();
+
+    await client.close();
+
```

Comment:
rather close in `finally` block, it is important to ensure the connection closes to avoid leaks on the server.